ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkReader.java

```diff
@@ -130,6 +139,36 @@ private Void readGenerator(String[] tokens) {
         return null;
     }
 
+    public AmplNetworkReader readBatteries() throws IOException {
+        read("_batteries", 7, this::readBattery);
+
+        return this;
+    }
+
+    private Void readBattery(String[] tokens) {
```

Comment:
Perhaps could we change "Function<String[], Void>" to "Consumer<String[]>" in read(...) method to avoid having this strange "Void" return type with "return null" statement ?

Reply:
I agree with your proposal. Il will create a new *first good issue* to refactor AmplNetworkReader.

Reply:
So we consider it is part of an other issue, and it is though non blocking right ?

Reply:
@GhilesA just followed the code structure made few monthes/years ago. To fix that, you also have to change for all other types of equipments. It more global than supporting battery in AMPL exporter.

Reply:
https://github.com/powsybl/powsybl-core/issues/754